clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -1579,8 +1585,15 @@ mlir::Value ScalarExprEmitter::VisitInitListExpr(InitListExpr *E) {
   if (E->hadArrayRangeDesignator())
     llvm_unreachable("NYI");
 
-  if (UnimplementedFeature::cirVectorType())
-    llvm_unreachable("NYI");
+  if (E->getType()->isVectorType()) {
+    SmallVector<mlir::Value, 16> Elements;
+    for (Expr *init : E->inits()) {
+      Elements.push_back(Visit(init));
+    }
+    return CGF.getBuilder().create<mlir::cir::VecValueOp>(
+        CGF.getLoc(E->getSourceRange()), CGF.getCIRType(E->getType()),
+        Elements);
+  }
```

Comment:
If we look at the traditional codegen for `VisitInitListExpr`, it's mostly already based on the LLVM types, therefore it's usually different enough from CIR that it's hard to follow the skeleton. In this case, we usually add asserts on `UnimplementedFeatures::XYZ` so that we capture the optimizations or extra peepholes that we don't yet support to make sure we can do proper lowering later. Can you add those here? Examples:

```
assert(UnimplementedFeature::scalableVectors());
assert(UnimplementedFeature::foldShuffleForTheSwizzle());
....
```

If relevant, do the same for `VisitArraySubscriptExpr`

Reply:
I couldn't figure out what the purpose of the `assert(UnimplementedFeature::xyz)` statements were.  With this explanation I think I get it.  It's to help catch places that should have been changed when a feature is finally implemented, but were forgotten.  I will try to pay attention to that and start adding them when appropriate.

Reply:
Yep. It's really helpful to track cases we cannot assert but still allows to track missing pieces for future work.